src/ContentRepository/Linq/SnExpression.cs

```diff
@@ -100,6 +100,7 @@ private static LucQuery BuildLucQuery(System.Linq.Expressions.Expression express
                     query.Top = childrenDef.Top;
                 if (childrenDef.Sort != null)
                     query.SetSort(childrenDef.Sort);
+                query.AllPages = childrenDef.AllPages;

```

Comment:
boolean's default is false, alike the inlinecount property, so a conditional evaluation here would add nothing here but a redundant operation